### ISSUE TITLE
AMP training clips scaled gradients before unscale_, causing instability / NaN

### ISSUE THREAD

**Issue by Sumter39** [ymy-k/Hi-SAM#36]
In train.py, the AMP training step does:
```
gradsclaler.scale(loss).backward()
torch.nn.utils.clip_grad_norm_(model.parameters(), max_norm=1.0, norm_type=2)
gradsclaler.step(optimizer)
```
But it does not call `gradsclaler.unscale_(optimizer)` before gradient clipping.
This means gradients are clipped while still scaled by GradScaler, which can distort updates and lead to training instability / NaN after several epochs.
